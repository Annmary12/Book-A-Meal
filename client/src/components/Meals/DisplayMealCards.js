import React from 'react';
import PropTypes from 'prop-types';
import NoMealSet from './NoMealSet.js';
import MealCard from '../MealCard/MealCardContainer.js';
import styles from './meals.scss';

const DisplayMealCards = ({
  meals
}) => (
  <div className={styles.grid}>
    {
      meals.length > 0 ?
      meals.map(meal => (
        <MealCard
          key={meal.id}
          meal={meal}
          action="edit"
        />
      )) :
      <NoMealSet />
    }
  </div>
);

DisplayMealCards.propTypes = {
  meals: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default DisplayMealCards;
