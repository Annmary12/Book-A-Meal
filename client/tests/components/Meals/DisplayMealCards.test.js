import React from 'react';
import DisplayMealCards from '../../../src/components/Meals/DisplayMealCards';

describe('DisplayMealCards component', () => {
  it('should match snapshot if meal array is empty', () => {
    const wrapper = shallow(
      <DisplayMealCards
        meals={[]}
        getMeals={() => {}}
        pagination={{}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
  it('should match snapshot if meal array is not empty', () => {
    const meals = [{
      id: 's324d-safd-da32',
      title: 'Beans and bread',
      img: 'picture.jpg',
      description: 'great meal'
    }]
    const wrapper = shallow(
      <DisplayMealCards
        meals={meals}
        getMeals={() => {}}
        pagination={{}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  describe('handleLoadMore method', () => {
    it('should call getMeals function from props with pagination info', () => {
      const getMeals = jest.fn();
      const wrapper = shallow(
        <DisplayMealCards
          getMeals={getMeals}
          meals={[]}
          pagination={{currentPage: 1}}
        />
      );

      wrapper.instance().handleLoadMore();
      expect(getMeals).toBeCalledWith({
        limit: 10,
        page: 2
      });
    })
  });
});