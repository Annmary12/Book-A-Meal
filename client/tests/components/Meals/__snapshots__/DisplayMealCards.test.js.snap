// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayMealCards component should match snapshot if meal array is empty 1`] = `
<InfiniteScroll
  dataLength={0}
  disableBrowserPullToRefresh={true}
  hasChildren={false}
  hasMore={false}
  initialLoad={false}
  loader={
    <Loader
      message="loading..."
      width={40}
    />
  }
  next={[Function]}
  pageStart={1}
  pullDownToRefreshContent={
    <h3>
      Pull down to refresh
    </h3>
  }
  pullDownToRefreshThreshold={100}
  releaseToRefreshContent={
    <h3>
      Release to refresh
    </h3>
  }
>
  <NoMealSet />
</InfiniteScroll>
`;

exports[`DisplayMealCards component should match snapshot if meal array is not empty 1`] = `
<InfiniteScroll
  dataLength={1}
  disableBrowserPullToRefresh={true}
  hasChildren={true}
  hasMore={false}
  initialLoad={false}
  loader={
    <Loader
      message="loading..."
      width={40}
    />
  }
  next={[Function]}
  pageStart={1}
  pullDownToRefreshContent={
    <h3>
      Pull down to refresh
    </h3>
  }
  pullDownToRefreshThreshold={100}
  releaseToRefreshContent={
    <h3>
      Release to refresh
    </h3>
  }
>
  <div
    className="grid"
  >
    <Connect(MealCard)
      action="edit"
      key="s324d-safd-da32"
      meal={
        Object {
          "description": "great meal",
          "id": "s324d-safd-da32",
          "img": "picture.jpg",
          "title": "Beans and bread",
        }
      }
    />
  </div>
</InfiniteScroll>
`;
